feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/presentation/staking/start/StartStakingLandingViewModel.kt

```diff
@@ -0,0 +1,260 @@
+import io.novafoundation.nova.common.utils.SpannableFormatter
+import io.novafoundation.nova.common.utils.clickableSpan
+import io.novafoundation.nova.common.utils.colorSpan
+import io.novafoundation.nova.common.utils.drawableSpan
+import io.novafoundation.nova.common.utils.flowOf
+import io.novafoundation.nova.common.utils.formatAsSpannable
+import io.novafoundation.nova.common.utils.formatting.formatFractionAsPercentage
+import io.novafoundation.nova.common.utils.setEndSpan
+import io.novafoundation.nova.common.utils.setFullSpan
+import io.novafoundation.nova.common.utils.toSpannable
+import io.novafoundation.nova.common.utils.withLoadingShared
+import io.novafoundation.nova.core.updater.UpdateSystem
+import io.novafoundation.nova.feature_staking_impl.R
+import io.novafoundation.nova.feature_staking_impl.data.StakingSharedState
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.StartStakingCompoundData
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.StartStakingInteractorFactory
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.model.PayoutType
+import io.novafoundation.nova.feature_staking_impl.presentation.StakingRouter
+import io.novafoundation.nova.feature_staking_impl.presentation.staking.start.model.StakingConditionRVItem
+import io.novafoundation.nova.feature_wallet_api.domain.model.Asset
+import io.novafoundation.nova.feature_wallet_api.presentation.formatters.formatPlanks
+import io.novafoundation.nova.feature_wallet_api.presentation.model.mapAmountToAmountModel
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import io.novafoundation.nova.runtime.state.assetWithChain
+import io.novafoundation.nova.runtime.state.chain
+import io.novafoundation.nova.runtime.state.chainAsset
+import java.math.BigDecimal
+import java.math.BigInteger
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.map
+import kotlin.time.Duration
+
+class StartStakingLandingViewModel(
+    private val stakingRouter: StakingRouter,
+    private val stakingSharedState: StakingSharedState,
+    private val resourceManager: ResourceManager,
+    private val updateSystem: UpdateSystem,
+    private val startStakingInteractorFactory: StartStakingInteractorFactory,
+    private val appLinksProvider: AppLinksProvider
+) : BaseViewModel(), Browserable {
+
+    private val startStakingInteractor = stakingSharedState.assetWithChain.map {
+        startStakingInteractorFactory.create(it.chain, it.asset, coroutineScope = this)
+    }
+
+    private val startStakingInfo = startStakingInteractor.flatMapLatest {
+        it.observeStartStakingInfo(stakingSharedState.chain(), stakingSharedState.chainAsset())
+    }
+
+    val isLoadingStateFlow = startStakingInfo.withLoadingShared()
+        .map { it.isLoading() }
+        .shareInBackground()
```

Comment:
Ususally we wrap data with `withLoadingShared` and use `when` to managa loading + show data. Why we decided to go with separate flow in this case?